main.cpp

```diff
@@ -58,17 +69,18 @@ struct List {
     }
 
     int pop_front() {
-        int ret = head->value;
-        head = head->next;
+        int ret = head.get()->value;
+        head = std::move(head.get()->next);
```

Comment:
```suggestion
        int ret = head->value;
        head = std::move(head->next);
```
`unique_ptr` 具有 `operator->`，不需要先 `get()` 再 `->`。